distributed/protocol/core.py

```diff
@@ -109,3 +115,5 @@ def _decode_default(obj):
     except Exception:
         logger.critical("Failed to deserialize", exc_info=True)
         raise
+    finally:
+        gc.enable()
```

Comment:
nit: it might be nice to only do this if [`gc.isenabled()`](https://docs.python.org/3/library/gc.html#gc.isenabled) was true before. Otherwise it would be very difficult for user code (like my profiling) to disable GC entirely.